src/docfx/lib/path/PackageUrl.cs

```diff
@@ -31,16 +34,8 @@ public PackageUrl()
 
         public PackageUrl(string url)
         {
-            if (UrlUtility.IsHttp(url))
-            {
-                Type = PackageType.Git;
-                (Url, Branch) = SplitGitUrl(url);
-            }
-            else
-            {
-                Type = PackageType.Folder;
-                Path = url;
-            }
+            Url = url;
+            OnDeserialized(s_streamingContext);
```

Comment:
if populate calls OnDeserialized, is it still necessary to call here?

Reply:
This one is called by  
![image](https://user-images.githubusercontent.com/42199316/65403057-cfdb4500-de04-11e9-9c08-d08f9b6c4019.png)
which share the same logic with OnDeserialized. And OnDeserialized needs to be manually invoked here.

Reply:
I see, streaming context could then be null

Reply:
I would very much like to. But it's a `struct` 😂.

Reply:
`StreamingContext.Context` is always null in both case